### PR TITLE
feat(uat): add test cases to T102 for request/response features

### PR DESCRIPTION
**Pull request by bgklika**
**Issue #, if available:**
Add test cases to T102 for request/response features

**Description of changes:**
- Add steps to set/reset clean session, keep alive, request response information, response topic tx/rx, correlation data tx/rx
- Add test cases for reponse topic, correlation data and request response information to T102 scenario outline
- Add response topic and correlation data to EventFilter
- Fix unhandled exception and tune logging in paho java client

**Why is this change necessary:**
Automatically test steps, clients and local broker with some new MQTT v5.0 features

**How was this change tested:**
Automatically run scenarios on CodeBuild 

**Test results:**
```
[INFO ] 2023-07-12 19:51:25.453 [main] StepTrackingReporting - Passed: 'GGMQ-1-T102-v5-sdk-java: As a customer, I can use new MQTT v5.0 features'
[INFO ] 2023-07-12 19:51:25.453 [main] StepTrackingReporting - Passed: 'GGMQ-1-T102-v5-mosquitto-c: As a customer, I can use new MQTT v5.0 features'
[INFO ] 2023-07-12 19:51:25.453 [main] StepTrackingReporting - Passed: 'GGMQ-1-T102-v5-paho-java: As a customer, I can use new MQTT v5.0 features'
[INFO ] 2023-07-12 19:51:25.453 [main] StepTrackingReporting - Passed: 'GGMQ-1-T102-v5-paho-python: As a customer, I can use new MQTT v5.0 features'
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java [hunk 1]
```diff
@@ -45,6 +45,10 @@ public String getConnectionName() {
 
     @Override
     public boolean isMatched(@NonNull EventFilter filter) {
+        return isMatchedPart1(filter) && isMatchedPart2(filter);
```
Comment (MikeDombo) [inline uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java:48]: why are there 2 parts?

Comment (MikeDombo) [reply uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java:48]: I do not see that they are being called individually

Comment (MikeDombo) [reply uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java:48]: Ignore it, that isn't useful in this instance.

Comment (MikeDombo) [reply uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java:48]: If you're going to just call method 1 and then method 2, I see no reason why you should split it. Keep the logic together.

Comment (bgklika) [reply uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java]: fixed in a7c07e944243724a0ffa89ecc8db1bf64c21e01d

**Review by MikeDombo — APPROVED** [review]

**Pull request merged**